Services/Init/classes/class.ilInitialisation.php

```diff
@@ -37,18 +37,23 @@ protected static function removeUnsafeCharacters()
 		// We do not need this characters in any case, so it is
 		// feasible to filter them everytime. POST parameters
 		// need attention through ilUtil::stripSlashes() and similar functions)
-		if (is_array($_GET))
-		{
-			foreach($_GET as $k => $v)
-			{
-				// \r\n used for IMAP MX Injection
-				// ' used for SQL Injection
-				$_GET[$k] = str_replace(array("\x00", "\n", "\r", "\\", "'", '"', "\x1a"), "", $v);
+		self::recursivelyRemoveUnsafeCharacters($_GET);
+	}
 
-				// this one is for XSS of any kind
-				$_GET[$k] = strip_tags($_GET[$k]);
+	protected static function recursivelyRemoveUnsafeCharacters(&$var) {
+		if (is_array($var)) {
+			foreach ($var as $k => $v) {
+				$var[$k] = self::recursivelyRemoveUnsafeCharacters($v);
```

Comment:
The key is not purged and may contain unsafe characters.
What do think about purging them, too?

Reply:
Done.

Reply:
Done.

Reply:
unset($var[$k]);
otherwise the old (unsanitized) keys remain in the array.